### ISSUE TITLE
Desktop Offering View in Automate

### ISSUE THREAD

**Issue by nfisher-chef** [chef/automate#2873]

**Comment by susanev** [comment]
>How do we guarantee this user only sees the Desktop Offering Dashboard view?

we can create a new access management role for this

**Comment by kmacgugan** [comment]
- Need to store more than 24 hours worth of data

This is already true excepting the ohai data, which we only store the latest data for. Will we be needing historical ohai data or just the current ohai data? Ohai data can be very large and managing historical data introduces new technical challenges. We store the rest of the chef-client run reports going back to configurable data retention settings.

**Comment by nfisher-chef** [comment]
> * Need to store more than 24 hours worth of data
> 
> This is already true excepting the ohai data, which we only store the latest data for. Will we be needing historical ohai data or just the current ohai data? Ohai data can be very large and managing historical data introduces new technical challenges. We store the rest of the chef-client run reports going back to configurable data retention settings.

yes, we will need historical Ohai data for certain values. Does that help? We don't need ALL of the data, but some.

**Comment by kmacgugan** [comment]
Yes, it helps clarify, do we know what those keys/values are? One of the challenges we've had with ohai data is that it is largely dynamic (including the keys! sometimes it uses values as keys!) so it poses a challenge to mapping in ES.

**Comment by nfisher-chef** [comment]
We are doing some discovery, so we will definitely know before we stop implementation! Data mapping should be done during our UX work.

**Comment by danielsdeleo** [comment]
### Product Decision: Top Errors API/UI ###

The UI mockup shows a "Top Errors" panel. We had a discussion of how this should work over time, in the context of several use-case scenarios and what is the most feasible to implement today. 

We agreed that for an initial implementation, we will aggregate the top errors from the node state index, with a hard-coded filter to include only errors from the past 7 days.

In other words, we are counting the errors on a node-wise basis where the most recent Chef Client run for that error ended in failure, and the most recent Chef Client run occurred within the past 7 day time period.

This design does not support trend information or historical analysis use-cases. Work to support those will be scheduled for later.
